src/kernel-install/50-depmod.install

```diff
@@ -33,6 +33,9 @@ case "$COMMAND" in
         exec depmod -a "$KERNEL_VERSION"
         ;;
     remove)
+        [ "$KERNEL_INSTALL_BLS_TYPE" -eq 2 ] && \
+            echo "BLS_TYPE=2, not removing modules.dep or associated files." && \
+            exit 0
```

Comment:
I think we can be a bit smarter when removing modules.dep & co: if --bls-type=1/2 was specified, then we need to check whether the entry of the other type exists. Otherwise we will always be leaving modules.dep installed when we use UKI, even if type-1 entry is not also present.